DependencyInjection/DoctrineExtension.php

```diff
@@ -126,6 +126,7 @@ protected function dbalLoad(array $config, ContainerBuilder $container)
 
         $container->setAlias('database_connection', sprintf('doctrine.dbal.%s_connection', $this->defaultConnection));
         $container->setAlias('doctrine.dbal.event_manager', new Alias(sprintf('doctrine.dbal.%s_connection.event_manager', $this->defaultConnection), false));
+        $container->setAlias('Doctrine\DBAL\Driver\Connection', new Alias('database_connection', false));
```

Comment:
I suggest putting this directly in dbal.xml, as it does not need to be dynamic based on the configuration